homeassistant/components/lifx/sensor.py

```diff
@@ -0,0 +1,254 @@
+            coordinator.get_rssi_unit_of_measurement()
+        )
+
+    @callback
+    def _async_update_attrs(self) -> None:
+        """Handle attribute updates."""
+        _LOGGER.debug(
+            "Updated RSSI sensor value for %s to %s",
+            self.coordinator.label,
+            self.coordinator.rssi,
+        )
+        self._attr_native_value = self.coordinator.rssi
+
+    @callback
+    async def async_added_to_hass(self) -> None:
+        """Enable fetch of RSSI data."""
+        _LOGGER.debug("Enabling RSSI sensor updates for %s", self.coordinator.label)
+        self.coordinator.update_rssi = True
+        await self.coordinator.async_request_refresh()
+        await super().async_added_to_hass()
+
+    @callback
+    async def async_will_remove_from_hass(self) -> None:
+        """Disable fetching RSSI data."""
+        _LOGGER.debug("Disabling RSSI sensor updates for %s", self.coordinator.label)
+        self.coordinator.update_rssi = False
+        await super().async_will_remove_from_hass()
+
+
+class LIFXUptimeSensorEntity(LIFXSensorEntity):
+    """Uptime sensor entity."""
+
+    def __init__(
+        self,
+        coordinator: LIFXSensorUpdateCoordinator,
+        description: SensorEntityDescription,
+    ) -> None:
+        """Initialise a LIFX Uptime sensor."""
+        self.entity_description = description
+        super().__init__(coordinator, description)
+
+    @callback
+    def _async_update_attrs(self) -> None:
+        """Update the uptime sensor attributes."""
+        self._attr_native_value = self.coordinator.uptime
+
+    @callback
+    async def async_added_to_hass(self) -> None:
+        """Enable updates for uptime sensor."""
+        self.coordinator.update_uptime = True
+        await self.coordinator.async_request_refresh()
+        await super().async_added_to_hass()
+
+    @callback
+    async def async_will_remove_from_hass(self) -> None:
+        """Disable updates for uptime sensor."""
+        self.coordinator.update_uptime = False
+        await super().async_will_remove_from_hass()
+
+
+class LIFXHevSensorEntity(LIFXSensorEntity):
+    """HEV Sensor entity."""
+
+    def __init__(
+        self,
+        coordinator: LIFXSensorUpdateCoordinator,
+        description: SensorEntityDescription,
+    ) -> None:
+        """Initialise a LIFX HEV sensor."""
+        super().__init__(coordinator, description)
+        self._attr_name = description.name
+        self.coordinator.update_method = cast(
+            Callable, f"self.coordinator._async_fetch_{description.key}"
+        )
+
+    @callback
+    def _async_update_attrs(self) -> None:
+        """Handle updating _attr values."""
+        self._attr_native_value = getattr(
+            self.coordinator, self.entity_description.key, None
+        )
+
+    @callback
+    async def async_added_to_hass(self) -> None:
+        """Enable fetch of HEV data."""
+        setattr(self.coordinator, f"update_{self.entity_description.key}", True)
```

Comment:
Please avoid `setattr`. Create a function on the coordinator instead

Reply:
Better would be to have the function return a callable with CALLBACK_TYPE and then wrap the result in `self.async_on_remove` so you can drop `async_will_remove_from_hass`